src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -233,6 +235,13 @@ private static ImmutableArray<DocumentInfo> CreateDocumentInfos(SourceFileInfo?[
                     Path.GetFileName(info.FilePath),
                     filePath: info.FilePath,
                     loader: info.Loader));
+
+                // If we successfully got something from SourceLink for this project then its nice to wait a bit longer
+                // if the user performs subsequent navigation
+                if (info.FromRemoteLocation)
+                {
+                    _sourceLinkEnabledProjects.Add(project.Id);
```

Comment:
So if this project just unloads, are we leaking the ID until later?

Reply:
But I guess we might already be doing that in other places since the pointer to the original workspace is also holding onto the project id?

Reply:
This is definitely an area I'm not confident of, but yes this is mirroring what happens in the decompilation side of things, both holding on to the original workspace, and project id. If there is a better approach here I'm happy to address in a follow up.